uwblueprint/supportive-housing
# [FIX] - Pagination Component redundant fetches

## GitHub Issue link

[Pagination Component - Prevent redundant / unintentional fetches](https://github.com/uwblueprint/supportive-housing/issues/111)



## Implementation description
* Created `fetchRecords` function to actually perform the get
* Added `onKeyUp` event handler to trigger fetch
* Modified `handleNumberInputChange` function to _only_ update the input value


## Steps to test
1. Seed log records (sufficient number such that we end up with some # of pages > 10)
2. Verify that changing the page number (without clicking outside of the text field / hitting enter) will NOT cause a refetch of log records
3. Verify that changing the page number and unfocusing the text field (whether through `tab` or clicking outside of it) will refetch log records
4. Verify that changing the page number and hitting enter will refetch the log records
5. Verify that hitting enter OR unfocusing the text field **without** changing the page number will NOT refetch log records


## What should reviewers focus on?
* The structure / call stack (might be a little confusing / messy) + there was some discussion around making this a reusable component in the future ; perhaps we might want to restructure the component within this PR itself?


## Checklist
